src/hooks/useAppBootstrap.functions.ts

```diff
@@ -0,0 +1,75 @@
+import Minilog from '@cozy/minilog'
+
+import { getErrorMessage } from '/libs/functions/getErrorMessage'
+
+const log = Minilog('useAppBootstrap.functions')
+
+interface OnboardingParams {
+  onboardUrl: string | null
+  fqdn: string | null
+}
+
+export const parseOnboardingURL = (
+  url: string | null
+): OnboardingParams | undefined => {
+  try {
+    if (!url?.includes('onboarding')) {
+      return undefined
+    }
+
+    const onboardingUrl = new URL(url)
+    const onboardUrl = onboardingUrl.searchParams.get('onboard_url')
+    const fqdn = onboardingUrl.searchParams.get('fqdn')
+
+    if (!onboardUrl && !fqdn) {
+      return undefined
+    }
+
+    return {
+      onboardUrl,
+      fqdn
+    }
+  } catch (error) {
+    const errorMessage = getErrorMessage(error)
+    log.error(
+      `Something went wrong while trying to parse onboarding URL data: ${errorMessage}`
+    )
+    return undefined
+  }
+}
+
+interface FallbackUrl {
+  mainAppFallbackURL: string | undefined
+  cozyAppFallbackURL: string | undefined
+}
+
+export const parseFallbackURL = (url: string | null): FallbackUrl => {
+  const defaultParse = {
+    mainAppFallbackURL: undefined,
+    cozyAppFallbackURL: undefined
+  }
+
+  if (url === null) {
+    return defaultParse
+  }
+
+  try {
+    const makeURL = new URL(url)
+    const fallback = makeURL.searchParams.get('fallback') ?? undefined
+    const isHome =
+      makeURL.pathname.startsWith('/home') ||
```

Comment:
there are a number of "magic strings" here and I'm not sure I'm too comfortable with that. For instance `"home"` already exists in `routes.js`. But what if tomorrow some other route name replaces `home`? Maybe we should use something like `mainPage` or `defaultPage`, just suggestions, trying to pass an idea

Reply:
What about this? https://github.com/cozy/cozy-flagship-app/compare/d65f4daf44c04c41d3ab436de7f31e9def13486c..c15517c19bb895b6b1682f74e176cd27f4d98b8b

I'm not sure of how this can change in the future.

For me the `/home` path in UniversalLinks is about some external app that requests to display `cozy-home`. It shouldn't have to worry about our MainApp concept. But maybe it should?
@Crash-- do you have any thoughts on this?

Reply:
I'm not sure to really understand the discussion here, but here is what I can say: 
- We'll be able to specify a new defaultApp (maybe what you call mainapp?) for a cozy. For now, this defaultApp is the Cozy-Home, it'll change in the future. 
- If you have universal link for `links.mycozy.cloud/flagship/toto`, then we should open the `toto` app. If we have link to `/flagship/home`, then we have to open the `home` (this is the slug so cozy-home). If we only have `/flagship`, then we should open the defaultApp.

Reply:
Yes that was the point: mainapp = defaultapp

My understanding is that today `flagship/home` will open the "main screen" (a.k.a CozyHomeScreen as it is named today)
But in the future we will have to check if `flagship/home` is the default app or if we should open it as a cozy-app. But maybe /home just won't exist on cozys with a different defaultapp?

Reply:
> But in the future we will have to check if flagship/home is the default app or if we should open it as a cozy-app (using CozyAppScreen). 

Oh, this is a nice remark, because, yes, /home will still be there even with a different defaultapp. Since this is the slug of an app.

Reply:
yeah that's why I think we should use semantic variables right now as it's very easily done, just to keep in mind that nothing is hardcoded nor it should be